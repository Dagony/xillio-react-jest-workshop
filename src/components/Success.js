import React, { Component } from 'react';

class Success extends Component {
    render() {
        return (
            <div>
                <h1 className="ui centered">Details Successfully Saved</h1>
                <span>Time: {Date.now()}</span>
            </div>
        );
    }
}

export default Success;
