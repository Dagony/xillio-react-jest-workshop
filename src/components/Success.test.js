import React from 'react';
import Success from './Success';
import {shallow} from 'enzyme';

describe('Success', () => {

    it('renders', () => {
        const wrapper = shallow(<Success />);
        // expect(wrapper.find('div')).toHaveLength(1);
        // expect(wrapper.find('div').text()).toEqual('Details Successfully Saved');
        expect(wrapper.find('div')).toMatchSnapshot();
    });

    it('snapshot test fails if content is not equal to previous run', () => {
        const wrapper = shallow(<Success />);
        expect(wrapper.text()).toMatchSnapshot();
    });

    describe('with mocked date', () => {
        let dateNowSpy;

        beforeEach(() => {
            dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => new Date('1984-10-06T02:20:00').getTime());
        });

        afterEach(() => {
            dateNowSpy.mockRestore();
        });

        it('snapshot test no longer fails if content is mocked, so equal to previous run', () => {
            const wrapper = shallow(<Success />);
            expect(wrapper.text()).toMatchSnapshot();
        });
    });
});