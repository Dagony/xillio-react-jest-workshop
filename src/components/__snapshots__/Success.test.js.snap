// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success renders 1`] = `ShallowWrapper {}`;

exports[`Success snapshot test fails if content is not equal to previous run 1`] = `"Details Successfully SavedTime: 1566827387172"`;

exports[`Success with mocked date snapshot test no longer fails if content is mocked, so equal to previous run 1`] = `"Details Successfully SavedTime: 465873600000"`;
